src/OrleansProviders/Streams/Memory/MemoryAdapterFactory.cs

```diff
@@ -79,6 +77,9 @@ public void Init(IProviderConfiguration providerConfig, string name, Logger log,
 
             // 10 meg buffer pool.  10 1 meg blocks
             bufferPool = new FixedSizeObjectPool<FixedSizeBuffer>(adapterConfig.CacheSizeMb, () => new FixedSizeBuffer(1 << 20));
+
+            this.serializer = this.serviceProvider.GetService<TSerializer>() ??
+                              (TSerializer)ObjectFactory.Value(this.serviceProvider, null);
```

Comment:
nit: I'd avoid duplication (which should always match) by extracting this to an internal static CreateSerializer(IServiceProvider) method, and then the MemoryBatchContainer<TSerializer> just calls MemoryAdapterFactory<TSerializer>.CreateSerializer